### ISSUE TITLE
GetSummaryInformation Function in BaiTranslator

### ISSUE THREAD

**Issue by jamesredpath** [kenwilcox/BankFileParsers#5]
Thank you very much for sharing your solution. The BAI translator is very helpful and well designed! 

We ran a file through with bank accounts of multiple currencies (ex: 1 USD and 4 CAD) and all of the summary records displayed as USD. I found that if I made the change below (commented lines represents the original code) then it would display the correct currency per bank account.
       
 public static List<SummaryHeader> GetSummaryInformation(TranslatedBaiFile data)
        {
            var ret = new List<SummaryHeader>();
            foreach (var group in data.Groups)
            {
                foreach (var account in group.Accounts)
                {
                    foreach (var fund in account.FundsTypes)
                    {
                        **var amount = BaiFileHelpers.GetAmount(fund.Amount, account.CurrencyCode, fund.TypeCode);**
                        // var amount = BaiFileHelpers.GetAmount(fund.Amount, group.CurrencyCode, fund.TypeCode);
                        ret.Add(new SummaryHeader()
                        {
                            Date = group.AsOfDateTime,
                            CreationDate = data.FileCreationDateTime,
                            SenderIdentification = data.SenderIdentification,
                            ReceiverIndetification = data.ReceiverIdentification,
                            FileIdentificationNumber = data.FileIdentificationNumber,
                            **CurrencyCode = account.CurrencyCode,**
                            // CurrencyCode = group.CurrencyCode,
                            CustomerAccountNumber = account.CustomerAccountNumber,
                            Amount = amount,
                            Count = fund.ItemCount,
                            FundType = fund.FundsType,
                            TypeCode = fund.Detail.TypeCode,
                            TypeDescription = fund.Detail.Description
                        });
                    }
                }
            }
            return ret;
        }

**Comment by kenwilcox** [comment]
Thanks for the kudos and the suggestion. I could add it myself, but I'd love to accept a pull request to give you credit for the fix. Is that something you're comfortable in doing?

**Comment by kenwilcox** [comment]
I was going to go ahead an make the changes myself. I noticed that in your pasted code GetAmount takes an additional parameter fund.TypeCode - were you using that for some reason?

**Comment by jamesredpath** [comment]
Apologies for the delay in replying back to you. I'm not sure how to do a pull request - but no worries, I don't need the credit for the fix. 

I added the fund.TypeCode parameter to GetAmount to check if the transaction type was a Debit or Credit. If the transaction type was a debit, I returned the amount * -1 to get a negative amount. Originally all of the amounts came out as positive, and I wanted the amount to reflect whether it was debit or credit.
